From 2ee8f66a5d1b25f9d037b44456e006a9972bb190 Mon Sep 17 00:00:00 2001
From: Hedayat Vatankhah <hedayatv@gmail.com>
Date: Wed, 19 Jun 2013 01:44:55 +0430
Subject: [PATCH 13/18] Issue with EFI detection in logger (#873207)

Fixed a bug introduced with the logger refactoring that caused EFI
detection to fail.

Resolves: rhbz#873207
---
 os-probes/mounted/x86/05efi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/os-probes/mounted/x86/05efi b/os-probes/mounted/x86/05efi
index 1e506fc..8a6288b 100755
--- a/os-probes/mounted/x86/05efi
+++ b/os-probes/mounted/x86/05efi
@@ -58,7 +58,11 @@ ret=1
 for test in /usr/lib/os-probes/mounted/efi/*; do
 	debug "running subtest $test"
 	if [ -f "$test" ] && [ -x "$test" ]; then
+		# we need results of subtest in stdout
+		orig_fd_res=$fd_result
+		export fd_result=1
 		entry=$("$test" "$mpoint/$efi")
+		export fd_result=$orig_fd_res
 		if [ -n "$entry" ]; then
 			debug "bootloader $entry found by subtest $test"
 			ret=0
-- 
2.5.5

